examples/access-control-migration/src/services/casbin.authorizer.ts

```diff
@@ -0,0 +1,72 @@
+import { Provider, inject } from '@loopback/core';
+import {
+  Authorizer,
+  AuthorizationContext,
+  AuthorizationMetadata,
+  AuthorizationRequest,
+  AuthorizationDecision,
+} from '@loopback/authorization';
+import * as casbin from 'casbin';
+
+// Class level authorizer
+export class CasbinAuthorizationProvider implements Provider<Authorizer> {
+  constructor(@inject('casbin.enforcer.factory') private enforcerFactory: (name: string) => casbin.Enforcer) { }
+
+  /**
+   * @returns authenticateFn
+   */
+  value(): Authorizer {
+    return this.authorize.bind(this);
+  }
+
+  async authorize(
+    authorizationCtx: AuthorizationContext,
+    metadata: AuthorizationMetadata,
+  ): Promise<AuthorizationDecision> {
+    const subject = this.getUserName(authorizationCtx.principals[0].id);
+    const object = authorizationCtx.resourceId ?? metadata.resource ?? authorizationCtx.resource;
+
+    const request: AuthorizationRequest = {
+      subject,
+      object,
+      action: (metadata.scopes && metadata.scopes[0]) || 'execute',
+    };
+
+    const allowedRoles = metadata.allowedRoles;
+
+    if (!allowedRoles) return AuthorizationDecision.ALLOW;
+    if (allowedRoles.length < 1) return AuthorizationDecision.DENY;
+
+    let allow: boolean = false;
+
+    for (const role of allowedRoles) {
+      const enforcer = await this.enforcerFactory(role);
+
+      const allowed_by_role = await enforcer.enforce(
```

Comment:
IIUC, is this where the authorizer calls casbin with enforcers to make decision?

Reply:
@agnes512 yep